includes/REST_API/Embed_Controller.php

```diff
@@ -137,15 +145,19 @@ public function get_proxy_item( $request ) {
 		$cache_key = 'web_stories_embed_data_' . md5( $url );
 
 		$data = get_transient( $cache_key );
-		if ( ! empty( $data ) ) {
-			if ( 'rest_invalid_story' === $data ) {
-				return new WP_Error( 'rest_invalid_story', __( 'URL is not a story', 'web-stories' ), [ 'status' => 404 ] );
-			}
-
-			$embed    = json_decode( $data, true );
-			$response = $this->prepare_item_for_response( $embed, $request );
 
-			return rest_ensure_response( $response );
+		if ( is_string( $data ) && ! empty( $data ) ) {
+			/**
+			 * Decoded cached embed data.
+			 *
+			 * @var array|null $embed
+			 */
+			$embed = json_decode( $data, true );
+
+			if ( $embed ) {
```

Comment:
If there json is invalid or this is empty, should we return a WP_error here?

Reply:
Then it would never be possible to clear that cache and fetch that data anew.

It makes sense to just fetch the data again if there's an error.